form3tech-oss/f1
# chore: rewrite prometheus test code

Changes:
 - remove `freeport` test dependency
 - To allow the run of tests in parralel replace the gobal mock instance running across all tests with per test server
 - Remove no longer needed global env setup for tests.
 - The new mock handler will now correctly fail tests on malformed data as it accepts `testing.T`
 - Separate the data store from the mock http server
 - Use typed store with locking to remove typecasting complexity
